server-spi-private/src/main/java/org/keycloak/services/cors/Cors.java

```diff
@@ -36,59 +33,67 @@
-    public static final String AUTHORIZATION_HEADER = "Authorization";
+    String ORIGIN_HEADER = "Origin";
+    String AUTHORIZATION_HEADER = "Authorization";
 
-    public static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
-    public static final String ACCESS_CONTROL_ALLOW_METHODS = "Access-Control-Allow-Methods";
-    public static final String ACCESS_CONTROL_ALLOW_HEADERS = "Access-Control-Allow-Headers";
-    public static final String ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers";
-    public static final String ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
-    public static final String ACCESS_CONTROL_MAX_AGE = "Access-Control-Max-Age";
+    String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    String ACCESS_CONTROL_ALLOW_METHODS = "Access-Control-Allow-Methods";
+    String ACCESS_CONTROL_ALLOW_HEADERS = "Access-Control-Allow-Headers";
+    String ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers";
+    String ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
+    String ACCESS_CONTROL_MAX_AGE = "Access-Control-Max-Age";
 
-    public static final String ACCESS_CONTROL_ALLOW_ORIGIN_WILDCARD = "*";
-    public static final String INCLUDE_REDIRECTS = "+";
+    String ACCESS_CONTROL_ALLOW_ORIGIN_WILDCARD = "*";
 
-    public static Cors add(HttpRequest request, ResponseBuilder response) {
+    static Cors builder() {
         KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
-        return session.getProvider(Cors.class).request(request).builder(response);
+        return session.getProvider(Cors.class);
     }
 
-    public static Cors add(HttpRequest request) {
-        KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
-        return session.getProvider(Cors.class).request(request);
-    }
-
-    public Cors request(HttpRequest request);
+    Cors builder(ResponseBuilder builder);
 
-    public Cors builder(ResponseBuilder builder);
+    Cors preflight();
 
-    public Cors preflight();
+    Cors auth();
 
-    public Cors auth();
+    Cors allowAllOrigins();
 
-    public Cors allowAllOrigins();
+    Cors allowedOrigins(KeycloakSession session, ClientModel client);
 
-    public Cors allowedOrigins(KeycloakSession session, ClientModel client);
+    Cors allowedOrigins(AccessToken token);
 
-    public Cors allowedOrigins(AccessToken token);
+    Cors allowedOrigins(String... allowedOrigins);
 
-    public Cors allowedOrigins(String... allowedOrigins);
+    Cors allowedMethods(String... allowedMethods);
 
-    public Cors allowedMethods(String... allowedMethods);
+    Cors exposedHeaders(String... exposedHeaders);
 
-    public Cors exposedHeaders(String... exposedHeaders);
+    /**
+     * Add the CORS headers to the current {@link org.keycloak.http.HttpResponse}.
+     */
+    void add();
 
-    public Cors addExposedHeaders(String... exposedHeaders);
```

Comment:
@rmartinc This is related to #26782 and I'm removing the method here as it is used from a single place. Now the `epxosedHeaders` is doing the same thing and adding additional "exposed" headers. 

The test should pass.

Reply:
@pedroigor Sorry for not answering before. I did exactly the same as you in the previous PR but @mposolda recommended to add the `addExposedHeaders` because it was a change of behavior. You can see Marek's comments in the [PR](https://github.com/keycloak/keycloak/pull/27015). I'm OK with it as I did the same as the first option. :smile: